crates/vizia_core/src/state/lens.rs

```diff
@@ -18,13 +19,64 @@ pub trait Lens: 'static + Clone {
     type Source;
     type Target;
 
-    fn view<O, F: FnOnce(Option<&Self::Target>) -> O>(&self, source: &Self::Source, map: F) -> O;
+    fn view<'b>(&self, source: &'b Self::Source) -> Option<LensValue<'_, 'b, Self::Target>>;
 
     fn name(&self) -> Option<&'static str> {
         None
     }
 }
 
+/// A type returned by `Lens::view()` which contains either a reference to model data or an owned value.
+pub enum LensValue<'a, 'b, T> {
+    /// A reference to local data
+    Local(&'a T),
+    /// A reference to model data
+    Borrowed(&'b T),
+    /// Owned data
+    Owned(T),
+}
+
+impl<T: Clone> Clone for LensValue<'_, '_, T> {
+    fn clone(&self) -> Self {
+        match self {
+            LensValue::Borrowed(v) => LensValue::Owned((*v).clone()),
```

Comment:
Could you just return a Borrowed here instead of cloning? I don't see what that accomplishes.
```suggestion
            LensValue::Borrowed(v) => LensValue::Borrowed(*v),
```

Reply:
Yes, good point. I'm not sure what I was thinking with that.